src/main/cpp/src/map_zip_with_utils.cu

```diff
@@ -0,0 +1,368 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "map_zip_with_utils.hpp"
+
+#include <cudf/column/column_factories.hpp>
+#include <cudf/detail/labeling/label_segments.cuh>
+#include <cudf/detail/null_mask.hpp>
+#include <cudf/lists/count_elements.hpp>
+#include <cudf/lists/gather.hpp>
+#include <cudf/lists/set_operations.hpp>
+#include <cudf/reduction.hpp>
+#include <cudf/table/experimental/row_operators.cuh>
+#include <cudf/table/table_view.hpp>
+#include <cudf/unary.hpp>
+#include <cudf/utilities/span.hpp>
+
+#include <thrust/scan.h>
+
+using namespace cudf;
+
+namespace spark_rapids_jni {
+
+// Taken from src/lists/utilities.cu
```

Comment:
3. This is a utility function that does not need external visibility.  It is implementation specific.  Please put this in a nested, unnamed namespace to force internal linkage.